lib/src/autosnapshotting/autosnapshotting.dart

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import '_snapshot.dart';
+import 'model.dart';
+
+Timer? _theTimer;
+final _takenSnapshots = <SnapshotRecord>[];
+late AutoSnapshottingConfig _config;
+bool _snapshottingIsInProgress = false;
+
+/// Enables auto-snapshotting, based on the value of ProcessInfo.currentRss (dart:io).
+///
+/// If autosnapshotting is already enabled, resets it.
+/// See [AutoSnapshottingConfig] for details.
+/// Use [stopAutoSnapshotOnMemoryOveruse] to stop auto-snapshotting.
+/// Snapshotting operation may cause a delay in the main thread.
+void autoSnapshotOnMemoryOveruse({
+  AutoSnapshottingConfig config = const AutoSnapshottingConfig(),
+}) {
+  stopAutoSnapshotOnMemoryOveruse();
+  _createFolderIfNotExists(config.folder);
+  _config = config;
+  _theTimer = Timer.periodic(config.interval, (_) {
+    if (_snapshottingIsInProgress) return;
+    _snapshottingIsInProgress = true;
+    _maybeTakeSnapshot();
+    _snapshottingIsInProgress = false;
+  });
+}
+
+void _createFolderIfNotExists(String folder) {
+  final dir = Directory(folder);
+  if (!dir.existsSync()) {
+    dir.createSync(recursive: true);
+  }
+}
+
+extension _SizeConversion on int {
+  int mbToBites() => this * 1024 * 1024;
+}
+
+bool _isFolderOversized() {
+  final folderSize = Directory(_config.folder)
```

Comment:
I haven't tried this myself, but can you use the `stat` function to get the directory size? https://api.dart.dev/stable/2.19.6/dart-io/FileSystemEntity/stat.html

Reply:
Tried. It returns much smaller size. Most likely size of the folder record.